corehq/apps/app_manager/decorators.py

```diff
@@ -146,5 +147,39 @@ def _no_conflict(*args, **kwargs):
     return _no_conflict
 
 
+def avoid_parallel_build_request(fn):
+    @wraps(fn)
+    def new_fn(app, comment, user_id, *args, **kwargs):
+        domain = app.domain
+        app_id = app.get_id
+        if _build_request_in_progress(domain, app_id):
+            return HttpResponse(_("There is already a version build in progress. Please wait."), status=400)
```

Comment:
This seems like a weird return statement to me since it is different to the type that you are returning below.
I think instead this should raise an exception, which you would then catch in the view and return this `HttpResponse` there